### PR TITLE
Size of boxes must now be nonnegative

### LINKED ISSUE #1106: Do not allow negative sizes for FloatBox, IntBox
When `FloatBox` or `IntBox` is initialised with a negative vector as size, the member variables `leftTop` and `size` need to be recalculated. 

Something like:
```
init: func@ (point: FloatPoint2D, vector: FloatVector2D) {
    if (vector x < 0) {
        point x += vector x
        vector x = -vector x
    }
    if (vector y < 0) {
        point y += vector y
        vector y = -vector y
    }
    this leftTop = point
    this size = vector
}
```

Decided in IRL discussion after closing https://github.com/cogneco/ooc-kean/pull/1054.

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1106

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @bettinaselig ?

File: source/geometry/FloatBox2D.ooc [hunk 1]
```diff
@@ -41,7 +41,18 @@ FloatBox2D: cover {
...
+		if (size x < 0) {
+			leftTop x += size x
+			size x = -size x
+		}
+		if (size y < 0) {
+			leftTop y += size y
+			size y = -size y
+		}
+		this leftTop = leftTop
+		this size = size
```
Comment (bettinaselig) [inline source/geometry/FloatBox2D.ooc]: According to https://github.com/cogneco/ooc-kean/issues/814 we want to avoid that local variables are called like class variables. I guess that is why we usually do not use the `.` for the input variables of the functions.

File: test/geometry/FloatShell2DTest.ooc [hunk 2]
```diff
@@ -86,9 +86,9 @@ FloatShell2DTest: class extends Fixture {
 			expect(increased top, is equal to(-1.0f) within(tolerance))
 			expect(increased right, is equal to(2.0f) within(tolerance))
 			expect(increased bottom, is equal to(3.0f) within(tolerance))
-			expect(decreased left, is equal to(1.0f) within(tolerance))
+			expect(decreased left, is equal to(0.0f) within(tolerance))
 			expect(decreased top, is equal to(1.0f) within(tolerance))
-			expect(decreased right, is equal to(0.0f) within(tolerance))
+			expect(decreased right, is equal to(1.0f) within(tolerance))
```
Comment (bettinaselig) [inline test/geometry/FloatShell2DTest.ooc]: Why this change?

Comment (marcusnaslund) [inline test/geometry/FloatShell2DTest.ooc]: Because, with the nonnegative size, `left` cannot be to the right of `right`.

Comment (bettinaselig) [inline test/geometry/FloatShell2DTest.ooc]: Ok.

Comment (marcusnaslund) [inline test/geometry/FloatShell2DTest.ooc]: Which makes sense, so this is good. :)

File: source/geometry/FloatBox2D.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/geometry/FloatBox2D.ooc]: Right you are. I will fix.

**Comment by marcusnaslund** [comment]
Updated.

**Comment by bettinaselig** [comment]
Now I am happy :-)
